src/platform/macos/mod.rs

```diff
@@ -427,14 +427,13 @@ impl OsIpcSender {
             // Zero out the last word for paranoia's sake.
             *((message as *mut u8).offset(size as isize - 4) as *mut u32) = 0;
 
-            let data_dest = shared_memory_descriptor_dest as *mut u8;
-            ptr::copy_nonoverlapping(data.as_ptr(), data_dest, data.len());
+            let mut data_dest = shared_memory_descriptor_dest as *mut u8;
+            let data_size = data.len();
+            let data_size_dest = data_dest as *mut usize;
+            *data_size_dest = data_size;
 
-            let mut ptr = message as *const u32;
-            let end = (message as *const u8).offset(size as isize) as *const u32;
-            while ptr < end {
-                ptr = ptr.offset(1);
-            }
+            data_dest = data_dest.offset(mem::size_of::<usize>() as isize);
+            ptr::copy_nonoverlapping(data.as_ptr(), data_dest, data_size);
```

Comment:
I understood your original question. I remain confused about how code that I wrote and committed in Message::size_of somehow did not make it to github both times I added it. I have verified that it is now present in the commits in this PR.